로그인 화면

EarlyHail opened the pull request:
### 구현의도

- 쿠키에 토큰이 없을 시 로그인 화면으로
   현재 React Route 기능은 사용 X
  로그인 페이지 외에 Public Router에 넣을 기능이 없으므로, MainLayout에서 바로 렌더링

- getCookie Util 함수 생성
  cookie 이름 (key)로 값을 찾아주는 함수

- 로고와 버튼으로 간단한 디자인 구현

### 기능 흐름도, 클래스 다이어그램(선택사항)
- 

### 사용된 기술(선택사항)
- 

### 리뷰 & 논의사항 & 궁금한점(선택사항)
- cookie값을 검사해서 로그인 페이지로 보내야하는데, 라우터를 이용할 방법이 도저히 생각이 안나 로그인 페이지를 렌더링했습니다.
- 애초에 MainLayout에서 로그인 여부를 확인하는게 이상해서
   1. 새로운 상위 컴포넌트를 추가
   2. App 컴포넌트에서 로그인 여부 파악
  으로 변경하는게 나을것 같은데 어떻게 생각하시는지 궁금합니다
- 디자인은 이게 최선이었습니다....

EarlyHail committed:
[feat] 토큰 없을 시 로그인 화면으로

- 쿠키에 토큰이 없을 시 로그인 화면으로
   현재 React Route 기능은 사용 X
  로그인 페이지 외에 Public Router에 넣을 기능이 없으므로, MainLayout에서 바로 렌더링

- getCookie Util 함수 생성
  cookie 이름 (key)로 값을 찾아주는 함수

EarlyHail committed:
[feat] 로그인 화면 디자인 추가

- 로고와 버튼으로 간단한 디자인 구현

EarlyHail committed:
[fix] OAuth url 변경

- OAuth url을 `localhost`에서 `panopticon-dev.gq`로 변경

junsushin-dev reviewed (approved):
프론트엔드 코드가 깔끔하네요. 고생 많으셨습니다! 😉
라우터를 이용한 로그인 여부 검사 관련해서는, [react-router-dom 공식 홈페이지의 예제](https://reactrouter.com/web/example/auth-workflow)를 참조하면 좋을 것 같습니다. 
(예전에 한번 해봤는데 callback 연계가 많아서 조금 어렵습니다...)

saeeng reviewed (approved):
좋습니다!!

juyoungpark718 commented on src/pages/Login.tsx:
```diff
@@ -21,8 +21,8 @@ const useStyles = makeStyles((theme) => ({
 const isDev: boolean = process.env.NODE_ENV === 'development';
-// const oauthURL: string = isDev ? 'http://panopticon-dev.gq/api/auth/github' : '/api/auth/github';
-const oauthURL: string = isDev ? 'http://localhost:3000/api/auth/github' : '/api/auth/github';
+const oauthURL: string = isDev ? 'http://panopticon-dev.gq/api/auth/github' : '/api/auth/github';
```
```suggestion
const OAUTH_URL: string = isDev ? 'http://panopticon-dev.gq/api/auth/github' : '/api/auth/github';
```
대문자로 하시는건 어떨까요??

juyoungpark718 reviewed (approved):
정말 고생하셨습니다 !! 👍

EarlyHail committed:
[fix] MainLayout 오류 수정, Login 변수이름 수정

- MainLayout 오류 수정
   conflict 해결 rebase과정에서 코드를 잘못 반영했던것 수정

- 환경변수 login url을 가져오는 변수 명 수정
